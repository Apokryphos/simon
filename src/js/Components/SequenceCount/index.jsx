import React from 'react';
import PropTypes from 'prop-types';

function SequenceCount(props) {
  return (
    <span className="sequence-count" title="Number of button presses in current sequence.">
      {props.count}
    </span>
  );
}

SequenceCount.propTypes = {
  count: PropTypes.number.isRequired,
};

export default SequenceCount;
