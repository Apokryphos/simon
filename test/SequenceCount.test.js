import React from 'react';
import { shallow } from 'enzyme';
import SequenceCount from 'Components/SequenceCount';

describe('SequenceCount rendering', () => {
  it('Renders OK', () => {
    const wrapper = shallow(<SequenceCount count={0} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.props('count') === 0);
  });
});
