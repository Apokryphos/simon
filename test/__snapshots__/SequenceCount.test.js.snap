// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SequenceCount rendering Renders OK 1`] = `
<span
  className="sequence-count"
  title="Number of button presses in current sequence."
>
  0
</span>
`;
